### ISSUE TITLE
Accessibility Insights: Element has children which are not allowed: igx-grid-header-row[tabindex], div[tabindex], button[aria-haspopup], button[tabindex], [role=navigation], [role=button], [role=switch], [role=combobox]

### ISSUE THREAD

**Issue by ipolo3** [IgniteUI/igniteui-angular#16661]
## Description  
Based on Accessibility Insights for Web, the following elements are not compliant: igxGrid

 * igniteui-angular version: 19.2.24
 * browser: Edge

## Steps to reproduce  

1. Install the Accessibility Insights extension for Edge
2. Create an example igx-grid table with columns, paginator and toolbar
3. Run the FastPass check from the notification Launch Pad

## Result  
What is the actual result after following the steps to reproduce?

We are seeing the following result:

<img width="743" height="331" alt="Image" src="https://github.com/user-attachments/assets/06197391-503e-4b0b-800d-762b9f1758e7" />

Title: WCAG 1.3.1: Ensure elements with an ARIA role that require child roles contain them (#igxGrid)
Tags: Accessibility, WCAG 1.3.1, aria-required-children

Issue: Ensure elements with an ARIA role that require child roles contain them (aria-required-children - https://accessibilityinsights.io/info-examples/web/aria-required-children)

Target application: CO+I CSP - https://coi-cspw.azurefd.net/supplystore

Element path: #igxGrid

Snippet: <igx-grid class="full-toolbar igx-grid" tabindex="0" role="grid" aria-colcount="59" id="igxGrid" aria-rowcount="1" style="width: 100%; display: grid; grid-template-rows: auto auto auto 1fr auto auto;">

Related paths: 
  igx-grid-header-row
  .igx-grid__tbody-content
  .igx-grid__tfoot > div
  .me-3
  #clearFilters-btn
  igx-page-nav
  .date-picker-wrapper
  button[name="btnColumnHiding"]
  button[title="Open exporter dropdown"]
  .h-100 > .menu-button.mdc-button--raised.mat-mdc-raised-button
  #mat-mdc-slide-toggle-2-button
  .input

How to fix: 
Fix any of the following:
  Element has children which are not allowed: igx-grid-header-row[tabindex], div[tabindex], button[aria-haspopup], button[tabindex], [role=navigation], [role=button], [role=switch], [role=combobox]

Environment: Microsoft Edge version 143.0.0.0

====

This accessibility issue was found using Accessibility Insights for Web 2.46.0 (axe-core 4.10.2), a tool that helps find and fix accessibility issues. Get more information & download this tool at http://aka.ms/AccessibilityInsights.

## Expected result  
What is the expected result after following the steps to reproduce?
There should not be failed instances

## Attachments  
Attach a sample if available, and screenshots, if applicable.  

![Image](https://github.com/user-attachments/assets/01f4e698-f384-4265-9467-e2bec6dd03a0)

**Comment by ChronosSF** [comment]
@ipolo3 , we are implementing some improvements but when the grid has certain features enabled such as the toolbar, groupby area or pager, as they are children of the `igx-grid` element, accessibility insights will complain as they are UI elements that are not compatible to what role="grid" expects.

Adding further changes in the structure of the grid to completely solve this are unfeasible as they'll be time consuming, potentially introduce breaking changes and not introduce any improvements in the actual accessibility of the grid. We are primarily focused on making the grid as usable as possible in real-world scenarios and infractions in the spec are less important to us than actual performance with readers.

If you would like for the app you are building to pass that check entirely and you are using any of the features that rely on non-compatible UI as listed by @viktorkombov  in the PR, you can use disable the default UI for those features and provide external templates that implement similar functionality. We have samples under https://www.infragistics.com/products/ignite-ui-angular/angular/components/grid/grid that can help you with this effort.
